services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/directory/DiagnosisKeysCountryDirectory.java

```diff
@@ -48,8 +49,11 @@
    */
   public DiagnosisKeysCountryDirectory(DiagnosisKeyBundler diagnosisKeyBundler,
       CryptoProvider cryptoProvider, DistributionServiceConfig distributionServiceConfig) {
-    super(distributionServiceConfig.getApi().getCountryPath(), ignoredValue ->
-        Set.of(distributionServiceConfig.getSupportedCountries()), Object::toString);
+    super(distributionServiceConfig.getApi().getCountryPath(), ignoredValue -> {
+      Set<String> countries = new HashSet<>(Set.of(distributionServiceConfig.getSupportedCountries()));
```

Comment:
There are two sets created here, whereas maybe only one could be used (Set.of())